pkgs/modules/rust/default.nix

```diff
@@ -1,61 +1,35 @@
-{ pkgs-unstable, lib, ... }:
+{ pkgs, lib, ... }:
 let
-  pkgs = pkgs-unstable;
-  cargoRun = pkgs.writeScriptBin "cargo_run" ''
-    if [ ! -f "$HOME/$REPL_SLUG/Cargo.toml" ]; then
-      NAME=$(echo $REPL_SLUG | sed -r 's/([a-z0-9])([A-Z])/\1_\2/g'| tr '[:upper:]' '[:lower:]')
-      ${pkgs.cargo}/bin/cargo init --name=$NAME
-    fi
-
-    ${pkgs.cargo}/bin/cargo run
-  '';
-  rust-version = lib.versions.majorMinor pkgs.rustc.version;
+  inherit (pkgs.fenix.stable) toolchain;
 in
 {
-  id = "rust-${rust-version}";
+  id = "rust-stable";
   name = "Rust Tools";
 
-  replit.packages = with pkgs; [
-    cargo
-    clang
-    rustc
-  ];
+  replit.dev.packages = with toolchain; [
+    toolchain
 
-  replit.dev.packages = with pkgs; [
-    rustfmt
-    rust-analyzer
+    pkgs.clang
```

Comment:
If we need this for the Rust, can we wrap rust instead so we don't leak all of clang into the environment?

Reply:
Okay if this is a followup since it already had this problem before.

Reply:
i don't think this makes sense. some rust tooling still relies on there being a c compiler in the environment eg during build scripts. sure, those will be available during the build scripts, but then it becomes invisible to the user which c compiler is being used.

Reply:
Okay, that sounds like a reasonable argument.